### DISCUSSION TITLE
Navigation profiles

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#19] [Q&A] [answered]
**Navigation profiles** are coming to Cruiser: customize navigation to YOUR driving style.

---

Routing settings (image 1):
- Select your **travel mode**: Car | Motorcycle | Bike | Foot
- Select your preferred **default speed** (average, not max)

Each **travel mode** and **selected speed** affect **all** navigation instructions differently.

---

Voice guidance settings (image 2):
- Select precisely which **navigation instructions** you want:
  - Distances: Large | Medium | Small
  - After turn points
  - Repeat instructions
- Adjust the **announcement time** (speak **turn now** / **arrival** instructions earlier or later)

---

You can keep the default settings and you will have the existing familiar navigation experience.
Or change them according to YOUR **driving style**. Test the results in the navigation simulation.

Coming soon in next Cruiser...

<img src="https://user-images.githubusercontent.com/3484020/213919468-1fc52d83-99af-4f27-b0a9-d7fae522eac5.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/213919470-175833f7-e4c6-4c07-89a2-55c68fdadb41.jpg" width="50%"/>

**Comment by Ciutnik** [comment]
It looks promising and even exciting! Will the motorcycle mode have the option to enable or disable unpaved roads depending on your preferences? The same applies to main or narrow roads.

**Reply by devemux86** [reply]
> enable or disable unpaved roads
> main or narrow roads

Such routing options depend on what is available from each routing engine.
e.g. BRouter (offline): more powerful is usually to change the profiles.

Currently there are the most common options (avoid motorways, toll roads, ferries, borders).
As several routers offer more options, I plan to enrich the routing options with more of them.

**Reply by Ciutnik** [reply]
OK, I understand. We are dependent on external tools. Either way, I'm very excited about Cruiser's development and hope for the future :-) Thanks!

**Reply by DolomitesTom** [reply]
Same here, sounds good

**Comment by Ciutnik** [comment]
And therefore, one more question comes to my mind: can this function be related to route segments (known to us otherwise) in the future? It would be brilliant.

**Reply by devemux86** [reply]
> can this function be related to route segments

Thanks for bringing it up!

One of Cruiser's next planned big features (the other is a _new universal offline search_)
is to use different routing options per route segment (regardless of the routing engine).

For example, then we could use straight line profile in all routing engines.

**Comment by DolomitesTom** [comment]
> Each travel mode and selected speed affect the navigation instructions differently.

How does the average speed then influences the road type? If Idon't want to use motorways, but mainly country roads and preferably not within cities, do I then need to set the average speed as low as possible? Is it the higher I set it, the more motorways/highways will be taken?

**Reply by devemux86** [reply]
- The selected **default / average speed** of profiles affects the trigger distance of voice announcements.
**Larger** speed -> **earlier** announcement of voice instructions, **smaller** speed -> **later** announcement.
- Another way to change the timing of voice announcements is to configure the "**Announcement time**".
Instead of fixed options, it is a scale with steps: **150%** (early) - **100%** (normal announce) - **25%** (late).

The default options work pretty much the way the voice navigation is now.

**Reply by DolomitesTom** [reply]
Thank you for clarifying voice guidance results.

What about routing influence with travel mode Motorcycle and default speed? If I don't want to use highways, but more small roads outside of cities? When are they preferrred by the routing engine? When setting the average speed low, 40 till 60? My usual average speed is between 70 and 90 km/h on side (secondary) roads. My guess is: if I would set it to 85 km/h, it would then prefer motorways/higways or main roads instead of small or secondary roads, am I right or is this a wrong guess?

**Reply by devemux86** [reply]
Now this is about the navigation and the voice instructions.

Routing depends on the available options of routing engines.
e.g. BRouter (offline) has very powerful [profiles](https://brouter.de/brouter/profiles2/) to customize.
Online routers may provide different routing options / vehicle.

Please see also my answer above: https://github.com/devemux86/cruiser/discussions/19#discussioncomment-4750084.
Routing options in Cruiser will be extended in the future.
And we will see what influence speed can play on them.

**Comment by devemux86** [comment]
Clarification (I edited the first post):

- The selected default / average **speed** affects **all** navigation instructions:
`In 1.2 km turn right`, `Turn right`, `Arrive at destination`, etc.
- The **announcement time** affects the **turn now** and **arrival** instructions:
`Turn right`, `Arrive at destination`, etc. (where "earlier" or "later" have meaning)

In the future the **announcement time** will also affect the nearby favorites / POI with distance (see https://github.com/devemux86/cruiser/discussions/9).

**Comment by devemux86** [comment, marked as answer]
**Navigation profiles** are available in [Cruiser 3.0.8](https://play.google.com/store/apps/details?id=gr.talent.cruiser) with selection:
- Car, Motorcycle, Bike, Foot
- Default speed
- Voice instructions
- Announcement time

For more details please see the [announcement](https://github.com/devemux86/cruiser/discussions/19#discussion-4782035) above.

**Comment by Ciutnik** [comment]
Thanks for the new version of Cruiser! 👍 
Do the new profiles only work with the OpenRouteService engine? I've changed the engine so many times that I'm already lost which was the default one...

**Reply by devemux86** [reply]
> Do the new profiles only work with the OpenRouteService engine?

All routing engines can calculate routes for car / motorcycle or bike or foot.

Except [CycleStreets](https://www.cyclestreets.net/) which is to plan routes specifically for bikes (fastest, balanced, quietest).

**Reply by Ciutnik** [reply]
> All routing engines can calculate routes for car / motorcycle or bike or foot.

In fact, although there seems to be no difference between a motorcycle and a car. Besides, when asking about profi, I was more referring to profile + avoidance combo (which does not work with every engine).

**Reply by devemux86** [reply]
> profile + avoidance combo (which does not work with every engine)

Avoid options are currently available by MapQuest, OpenRouteService, Valhalla.

Note that all of these are highly customizable with BRouter (offline) profiles.
BRouter profiles can be best tested on the [BRouter website](https://brouter.de/brouter-web).

**Comment by Leeb64** [comment]
Wonderful!
You can now select the "moped profile" at the online Brouter (motorcycle/short).
Much faster than offline with the Brouter.

**Comment by devemux86** [comment]
> You can now select the "moped profile" at the online Brouter (motorcycle/short).

Current mapping of Cruiser options to BRouter profiles:
- Car
  - Fastest -> car-fast
  - Shortest -> car-eco
- Motorcycle (same as car)
- Bike
  - Fastest -> fastbike
  - Shortest -> trekking
- Foot -> shortest

If you need different BRouter profiles, we can discuss it.
You can find all BRouter profiles [here](https://brouter.de/brouter/profiles2/).

---

BRouter [moped](https://brouter.de/brouter/profiles2/moped.brf) profile mentions (is it true?):
```
# Moped-Routing is experimantal !!!
#
# DO NOT USE FOR ACTUAL NAVIGATION
#
# Turn restrictions are missing, leading to wrong routes
```

**Comment by Leeb64** [comment]
The motorcycle profile/short brought sometimes the same results as the moped profile in the offline brouter.-Not quite the same-
It would be nice I you could choose this moped profile as a motorcycle profile!->Smaller roads are preferred.
However, the time calculation of the moped profile is not correct....

**Reply by devemux86** [reply]
I'm thinking if BRouter is selected in Cruiser settings,
to offer the complete collection of BRouter profiles.

> However, the time calculation of the moped profile is not correct....

The times in the calculated routes are provided by the routing services.

Now that we can change the default speed / profile in Cruiser settings,
we could have the times to be calculated based on the selected speed?

**Reply by Leeb64** [reply]
Good idea!

**Comment by Leeb64** [comment]
Well, the problem with the wrong time calculation would be solved here....

**Comment by Cer-o** [comment]
Hello everyone, I try to follow all the conversations but due to my little experience it is not easy for me.

[I get lost when you talk about a router (I have BRouter installed for when I was doing offline routes with kurviger)]

May I know which is the "base" configuration of Cruiser online and which is the best to do offline?

**Reply by devemux86** [reply]
> which is the "base" configuration of Cruiser online

Do you mean the online routing?

> which is the best to do offline

- GraphHopper (offline) routing is inside countries (it contains speed limits and street names)
- BRouter (offline) routing can cross borders (it doesn't contain speed limits and street names)
(and it allows to customize its [profiles](https://brouter.de/brouter/profiles2/))

**Reply by Cer-o** [reply]
Yes, I am not sure how to choose between online or offline routing and the differences between the different types

**Reply by devemux86** [reply]
> how to choose between online or offline routing

The routing provider can be selected in "Settings | Routing | Routing".

There are 2 offline:
- BRouter (offline)
- GraphHopper (offline)

The others are online.

Regarding their differences, all can calculate routes for car / motorcycle, bike, foot.
Except [CycleStreets](https://www.cyclestreets.net/) which is specifically for bikes.
OpenRouteService and Valhalla offer some avoid options.
But you should try them out to see if any are useful to you.

**Reply by Cer-o** [reply]
Is there much difference between BRouter or GraphHopper online and offline?

**Reply by devemux86** [reply]
> Is there much difference between BRouter or GraphHopper online and offline?

Both BRouter should be similar.

GraphHopper (offline) is a special version for mobile, so it might be a little different.

**Reply by Cer-o** [reply]
Thank you

**Comment by DolomitesTom** [comment]
> Current mapping of Cruiser options to BRouter profiles:
    Car
        Fastest -> car-fast
        Shortest -> car-eco
    Motorcycle (same as car)

just tested in 3.0.8

Have set to Motorcycle shortest and played with different speed settings for motorcycle between 30 and 90 k m/h

There is no difference in route calculation. It is always calculated according to the setting brouter app.

When switching to quickest routing type in Cruiser, I would expect it to use highways, but it doesn't because I have set car_eco in brouter app and used this setting:
assign avoid_motorways     = true  # %avoid_motorways% | Avoid motorways | boolean

The calculated route contains no highways. Only when I switch to car_fast in brouter, then highways are used.

This means the cruiser settings do not overrule the settings in brouter. I guress this has to work before different routing options per segment can work, right?

**Reply by devemux86** [reply]
> This means the cruiser settings do not overrule the settings in brouter.

That is right.

Cruiser profiles [Car, Motorcycle, Bike, Foot] affect the routing,
in the sense that the selected routing engine calculates different routes for each vehicle.

**Reply by DolomitesTom** [reply]
ok, but then this doesn't work with BROUTER offline. As described, it has no effect: When selection AUTO and quickest (which is from your description equal to Motorcycle and quickest), it doesn't work. Brouter offline calculate always what I have set there. In this case car_eco which doesn't use quick roads (highway), even if set to Auto and quickest in Cruiser.

**Reply by WolfgangZ42** [reply]
> it doesn't work.

That's not true. I've defined BRouter (offline), Auto and fastest and get this:

<img src="https://user-images.githubusercontent.com/121806238/214815086-fc998716-20b9-463d-805b-b70e3eedce52.png" width="200"/>

Then I changed just fastest to shortest and the result is:

<img src="https://user-images.githubusercontent.com/121806238/214815136-f315cda9-5322-4fbb-8c96-6814a7269397.png" width="200"/>

**Reply by devemux86** [reply]
If BRouter (offline) app is unchanged, then it is used as mentioned [above](https://github.com/devemux86/cruiser/discussions/19#discussioncomment-4779322).

We can assign BRouter profiles to Cruiser settings "Travel mode" + "Route type".
Also we can change how BRouter profiles calculate routes.

Then these changed / assigned BRouter (offline) profiles are used.

**Reply by devemux86** [reply]
> I forgot to resize the pictures

It can be done automatically, see https://github.com/devemux86/cruiser/discussions/27.

**Comment by DolomitesTom** [comment]
> We can assign BRouter profiles to Cruiser settings "Travel mode" + "Route type".

Yes please

> Also we can change how BRouter profiles calculate routes.

yes, please

> Then these changed / assigned BRouter (offline) profiles are used.

Not sure if I understand this correctly. So you need to do a change in next cruiser version and then the Cruiser settings are used with Brouter offline calculation independent from when the brouter app settings have been changed recently?

**Reply by devemux86** [reply]
When BRouter (offline) app is unchanged with the defaults, then it is used like this:
(see the BRouter app image, it is similar to BRouter website)
- Car
  - Fastest -> car-fast
  - Shortest -> car-eco
- Motorcycle (same as car)
- Bike
  - Fastest -> fastbike
  - Shortest -> trekking
- Foot -> shortest

You can change the mapping inside BRouter app (using Server mode) and also edit the BRouter profiles.
Then such customizations have priority and are used by BRouter app.

<img src="https://user-images.githubusercontent.com/3484020/214839654-4b49e3d9-2b1d-426f-90fa-9de4b12de81a.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
And how do I delete such settings in Brouter in order to use the Cruiser settings? Or how can It set the defaults in Brouter offline?

**Reply by devemux86** [reply]
As with all Android apps, you can usually clear their data in the Android app settings.
In the case of BRouter, this may also delete the downloaded routing data and edited profiles.
(like uninstall BRouter and install it again)

Or just reassign the BRouter profiles based on the image above (using Server mode).

**Comment by DolomitesTom** [comment]
Ahhh ok , now i understand the first time how Brouter works 🌝. Thanks, it works now. Now the Route calculation by segments can come 😂.

**Comment by Leeb64** [comment]
Back to the timing issue with the Brouter mopedprofile...
For me, changing the standard speed does not change the driving time. Maybe I'm too stupid?
But I found out how the Brouter calculates the right time:
Set the value for "assign validForBikes" from 1 to 0 in the Moped.brf file.

**Reply by devemux86** [reply]
> changing the standard speed does not change the driving time

Profiles in Cruiser settings currently set the vehicle in route calculation and the timing in navigation instructions.
The route duration is still calculated by the selected routing service.

I suggested [above](https://github.com/devemux86/cruiser/discussions/19#discussioncomment-4780090) to display route duration: from routing service or using the selected speed.

**Comment by Leeb64** [comment]
A large selection of BRouter profiles is available here:https://github.com/poutnikl/Brouter-profiles/tree/master/CarProfiles
With a little perseverance, the profiles can still be customized.

**Comment by 1maxi2** [comment]
Gibt es in Cruiser nicht die Möglichkeit wie in kurviger zu wählen : kurvig, extra kurvig, schnellste und so weiter ?

**Reply by devemux86** [reply]
Cruiser can **load offline kurviger routes** (not Kurviger routing).

Also see this discussion for more details: https://github.com/devemux86/cruiser/discussions/33#discussioncomment-4933915.

**Reply by DolomitesTom** [reply]
Nein, dafür viele andere Features und weitere Verbesserungen. BRouter offline ist sehr gut geeignet, völlig unabhängig von Internet Bandbreite oder Verfügbarkeit Routen anzupassen und zu navigieren. Nur diese funktioniert länderübergreifend. Die meisten dürften auch Kurviger Web benutzen zur Planung. Dort muss man dann eben mehr shaping Punkte setzen und in der kurviger Datei exportieren. Mit der BRouter App kann man die Grundeinstellugen anpassen und die Routingkarten runterladen. Mit Cruiser die Navigationskarten, POIs, Kartenstile und Ebenen. Dann die Kurviger Routen Datei in Cruiser importieren und los geht's. In weiteren Versionen soll Cruiser unterschiedliche Routingoptionen nach Abschnitten implementieren.
